phoenix/src/communication.rs

```diff
@@ -1,203 +1,205 @@
+                                                        } else {
+                                                            info!("Recovery manager not initialized.");
+                                                        }
+                                                    });
 
-        rx.listen();
-        // setup interrupts
+                                                    // COMMAND_CHANNEL.send(command_data).await;
+                                                    // info!("Received Deploy Drogue command: {:?}", deploy_drogue);
+                                                }
+                                                messages_prost::command::command::Data::DeployMain(deploy_main) => {
+                                                    RECOVERY_MANAGER.lock(|cell| {
+                                                        info!("Boom boom");
+                                                        // *cell.borrow_mut() = Some(recovery_manager);
+                                                        if let Some(recovery_manager) = cell.borrow_mut().as_mut() {
+                                                            recovery_manager.arm();
+                                                                recovery_manager.fire_main();
+                                                            recovery_manager.disarm();
+                                                        } else {
+                                                            info!("Recovery manager not initialized.");
+                                                        }
+                                                    });
+                                                    // info!("Received Deploy Main command: {:?}", deploy_main);
+                                                    // COMMAND_CHANNEL.send(command_data).await;
 
-        RadioDevice {
-            transmitter: tx,
-            receiver: PeekReader::new(rx),
+                                                }
+                                                messages_prost::command::command::Data::PowerDown(power_down) => {
+                                                    // info!("Received Power Down command: {:?}", power_down);
+                                                }
+                                                messages_prost::command::command::Data::RadioRateChange(rate_change) => {
+                                                    // info!("Received Radio Rate Change command: {:?}", rate_change);
+                                                }
+                                            }
+                                        }
+                                        // Handle Command
+                                    }
+                                }
+                            }
+                        } else {
+                            info!("Failed to decode radio frame.");
```

Comment:
Unlikely to happen but maybe we could log the specific decode error if its malformed.

```
match messages_prost::radio::RadioFrame::decode_length_delimited(&mut &msg.message[..]) {
Ok...
Err...
}
```